src/Core/Geometry/MeshPrimitives.cpp

```diff
@@ -3,6 +3,8 @@
 #include <Core/Math/Math.hpp> // areApproxEqual
 #include <Core/Types.hpp>
 
+#include <Engine/Data/Mesh.hpp>
```

Comment:
Use only Core data structure here. The Core Library should not be dependent on Engine